src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -725,6 +725,10 @@ private static void CreatePart(ZipArchive package, string path, Stream sourceStr
             string entryName = CreatePartEntryName(path);
 
             var entry = package.CreateEntry(entryName, CompressionLevel.Optimal);
+            if(File.Exists(path))
```

Comment:
Is this check needed? It looks like it opens the stream below without checking so the path must always exist.

Reply:
I think it is. The stream is create with an entryName not with the path.  It is my understanding that WriteFiles calls CreatePart an not always with Files that come from disk see comment in [WriteFiles](https://github.com/pepone/NuGet.Client/blob/7dfa9a17126030dd71a98109c97e41f495981664/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs#L569)

Reply:
Without the check there are some test failures

```
 System.ArgumentOutOfRangeException : The DateTimeOffset specified cannot be converted into a Zip file timestamp.
Parameter name: value
Stack Trace:
   at System.IO.Compression.ZipArchiveEntry.set_LastWriteTime(DateTimeOffset value)
   at NuGet.Packaging.PackageBuilder.CreatePart(ZipArchive package, String path, Stream sourceStream) in D:\NuGet.Cli
ent\src\NuGet.Core\NuGet.Packaging\PackageCreation\Authoring\PackageBuilder.cs:line 728
   at NuGet.Packaging.PackageBuilder.WriteFiles(ZipArchive package, HashSet`1 filesWithoutExtensions) in D:\NuGet.Cli
ent\src\NuGet.Core\NuGet.Packaging\PackageCreation\Authoring\PackageBuilder.cs:line 591
   at NuGet.Packaging.PackageBuilder.Save(Stream stream) in D:\NuGet.Client\src\NuGet.Core\NuGet.Packaging\PackageCre
ation\Authoring\PackageBuilder.cs:line 347
[xUnit.net 00:00:02.4621274]     NuGet.Packaging.Test.PackageBuilderTest.CreatePackageUsesV5SchemaNamespaceIfReferenc
esTargetFramework [FAIL]
   at NuGet.Packaging.Test.PackageBuilderTest.CreatePackageUsesV5SchemaNamespaceIfDevelopmentDependency() in D:\NuGet
.Client\test\NuGet.Core.Tests\NuGet.Packaging.Test\PackageBuilderTest.cs:line 1003```

Reply:
I see what you mean, looks like these files have gone through a lot of hoops before they get here.

Given that, I think the LastWriteTime should be added to IPackageFile
`DateTimeOffset LastWriteTime { get; }`

When the file is read from disk it should also copy in the LastWriteTime. Without that it could up getting the wrong file when trying to use the path in the nupkg to find the path on disk.